src/ORM/BehaviorRegistry.php

```diff
@@ -58,9 +58,22 @@ class BehaviorRegistry extends ObjectRegistry
     /**
      * Constructor
      *
-     * @param \Cake\ORM\Table $table The table this registry is attached to
+     * @param \Cake\ORM\Table $table The table this registry is attached to.
      */
-    public function __construct(Table $table)
+    public function __construct(Table $table = null)
```

Comment:
Could you remove the typehint here? I think I can use the fact that we have the setter to make the constructor more free to use for others